source/common/ssl/context_config_impl.cc

```diff
@@ -308,13 +308,12 @@ ServerContextConfigImpl::ServerContextConfigImpl(
 
         return ret;
       }()) {
-  // TODO(PiotrSikora): Support multiple TLS certificates.
   if ((config.common_tls_context().tls_certificates().size() +
        config.common_tls_context().tls_certificate_sds_secret_configs().size()) == 0) {
     throw EnvoyException("No TLS certificates found for server context");
-  } else if ((config.common_tls_context().tls_certificates().size() +
-              config.common_tls_context().tls_certificate_sds_secret_configs().size()) > 1) {
-    throw EnvoyException("A single TLS certificate is required for server contexts");
+  } else if (!config.common_tls_context().tls_certificates().empty() &&
+             !config.common_tls_context().tls_certificate_sds_secret_configs().empty()) {
+    throw EnvoyException("Static and dynamic TLS certificates may not be mixed in server contexts");
```

Comment:
Nit: this isn't really "static" and "dynamic", since LDS/CDS and SDS can be both "static" and "dynamic".

Maybe just say that (i.e. `LDS/CDS and SDS TLS certificates may not be mixed in server contexts`) or is it too nitty-gritty detail for operators?

Reply:
I think "Non-SDS and SDS TLS certificates may not be mixed.." is clearer here, as LDS/CDS doesn't refer to bootstrap static config.

Reply:
[secrets](https://github.com/envoyproxy/envoy/blob/25b143c9813671c320561f8b0b8b31d7e0a3dac4/api/envoy/config/bootstrap/v2/bootstrap.proto#L48) (i.e. "static" SDS) can be also in the bootstrap static config.

But yeah, I agree that it sounds cleaner.